### PR TITLE
docs: add documentation for story indexers

### LINKED ISSUE #20478: [Bug]: [7.0.0-beta.20]: Unable to change story loading globing parttern for something else than `stories`
### Describe the bug

Trying to update the story loading configuration with a globing pattern that doesn't contain the keyword `stories`. 
After updating that configuration, Storybook won't start and results into the following error:
```
Error: No matching story indexer found for `src/components/storybook-example/ExampleButton.[customName].tsx`
```

### To Reproduce

Based on the following configuration that is working :

```shell
├── .storybook
│   └── main.ts
├── src
├── └── storybook-example
└─────── └── ExampleButton.stories.tsx
```

```tsx
// .storybook/main.ts

import { ConfigEnv, mergeConfig } from 'vite';
import svgr from 'vite-plugin-svgr';
import { aliases } from '../src/aliases';

export default {
    stories: ['../src/**/*.mdx', '../src/**/*.stories.@(jsx|tsx)'],
    addons: ['@storybook/addon-links', '@storybook/addon-essentials', '@storybook/addon-interactions'],
    framework: {
        name: '@storybook/react-vite',
        options: {},
    },
    docs: {
        autodocs: 'tag',
    },
    async viteFinal(config: ConfigEnv) {
        return mergeConfig(config, {
            plugins: [svgr()],
        });
    },
};
```

1) Update the `stories` entry for something different, like follow:

```ts
stories: ['../src/**/*.mdx', '../src/**/*.storiestest.@(jsx|tsx)'],
```

2) Rename `ExampleButton.stories.tsx` to `ExampleButton.storiestest.tsx`

3) Restart Storybook and the following error will appear:

```
Error: No matching story indexer found for /Users/xxx/Documents/myproject/apps/client/src/components/storybook-example/ExampleButton.storiestest.tsx
    at StoryIndexGenerator.extractStories (/Users/xxx/Documents/myproject/node_modules/.pnpm/@storybook+core-server@7.0.0-beta.20/node_modules/@storybook/core-server/dist/index.js:10:10111)
```

### System

```shell
MacOS Monterey 12.5.1
pnpm version 7.17.1
```

### Additional context

- The project is a monorepo using `pnpm` and its workspaces features.
- Tried to clear the `node_modules/.cache` folder after updating the configuration.

### PR DESCRIPTION
**Pull request by sheriffMoose**
Issue: #20478 

## What I did

- [x] Added documentation for story indexers to the sidebar-and-urls doc.

## How to test

- [x] N/A

## Checklist

- [x] Make sure to add/update documentation regarding your changes

#### Maintainers

- [x] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: docs/configure/sidebar-and-urls.md [hunk 1]
```diff
@@ -128,3 +128,55 @@ Additionally, if you customize your Storybook to load your stories based on a [c
...
+
+<CodeSnippets
+  paths={[
+    'common/storybook-auto-docs-mdx-docs-docs-only-page.mdx.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+Typically Storybook provides indexing capabilities for files that end with `.story|stories.js|ts|jsx|tsx` and `.stories.mdx`. If you feel the need to include stories that have different naming convention, e.g. [`20478`](https://github.com/storybookjs/storybook/issues/20478), you will need to introduce a new story indexer.
```
Comment (shilman) [inline docs/configure/sidebar-and-urls.md:160]: We also index `.mdx` by default here:

https://github.com/storybookjs/storybook/blob/next/code/addons/docs/src/preset.ts#L135-L149

File: docs/snippets/common/csf-3-example-title.js.mdx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+```js
+// CSF 3
+const meta: Meta<Button> = {
```
Comment (shilman) [inline docs/snippets/common/csf-3-example-title.js.mdx:3]: Should this be `.ts.mdx`?

File: docs/configure/sidebar-and-urls.md [hunk 3]
```diff
@@ -128,3 +128,55 @@ Additionally, if you customize your Storybook to load your stories based on a [c
...
+
+<CodeSnippets
+  paths={[
+    'common/storybook-main-csf-indexer.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+Another example for indexing `.stories|story.mdx` files is here as well.
```
Comment (shilman) [inline docs/configure/sidebar-and-urls.md:172]: We should probably update the default to index `.story.mdx` as well to make it consistent with other parts of the code. WDYT about including your `.md` example here instead?

**Review by shilman — APPROVED** [review]
Thank you so much for writing this up. I'm sure @jonniebigodes will have thoughts on this as well, but I made some minor suggestions below for your consideration.

**Comment by stof** [comment]
shouldn't this be mentioned in the upgrade guide too ?

**Pull request merged**